File: surf-chat/src/main/java/dev/slne/surf/social/chat/listener/PlayerQuitListener.java
package dev.slne.surf.social.chat.listener;

import dev.slne.surf.social.chat.service.ChatHistoryService;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

public class PlayerQuitListener implements Listener {
  @EventHandler
  public void onQuit(PlayerQuitEvent event) {
    ChatHistoryService.getInstance().clearInternalChatHistory(event.getPlayer().getUniqueId());
  }
}

File: surf-chat/src/main/java/dev/slne/surf/social/chat/listener/PlayerQuitListener.kt
package dev.slne.surf.social.chat.listener

import dev.slne.surf.social.chat.provider.ChannelProvider
import dev.slne.surf.social.chat.service.ChatHistoryService
import dev.slne.surf.social.chat.service.ChatReplyService
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

class PlayerQuitListener : Listener {
    @EventHandler
    fun onQuit(event: PlayerQuitEvent) {
        ChatHistoryService.clearInternalChatHistory(event.player.uniqueId)
        ChannelProvider.handleQuit(event.player)
        ChatReplyService.clear(event.player.uniqueId)
    }
}
